### PR TITLE
chore: disabling static color hex stylelint rule for specific files

### LINKED ISSUE #23097: Configure Linting to Ignore Specific Files for 'color-no-hex' Rule
### What is this about?

To maintain the effectiveness of our linting process and prevent lint warning blindness, we propose updating our lint configuration to ignore specific files from the 'color-no-hex' rule. This approach aims to ensure that developers focus on meaningful lint warnings that are relevant to the main application, while acknowledging that certain external or standalone parts of our project (e.g., migration dashboards, documentation) may not strictly adhere to the design token system.

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

Scope: Update the lint configuration (stylelint.config.js or equivalent) to exclude specific files or directories from the 'color-no-hex' rule.
Files to Ignore: Identify and list the files or directories where static hex color values are permissible due to their standalone nature or external usage.

### Acceptance Criteria

The lint configuration is successfully updated to exclude the identified files from the 'color-no-hex' rule.
Developers no longer receive 'color-no-hex' lint warnings for the specified files, allowing them to concentrate on addressing warnings that impact the main application.
The overall quality and consistency of the codebase are maintained, with design tokens being used where applicable.

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**

This PR addresses the need to refine our linting process by excluding specific files from the 'color-no-hex' rule, as outlined in [Issue #23097](https://github.com/MetaMask/metamask-extension/issues/23097). The motivation behind this change is to prevent lint warning blindness among developers by focusing on meaningful lint warnings that are relevant to the main application. By updating our lint configuration to ignore certain files where static hex color values are permissible, we aim to maintain the effectiveness of our linting process while acknowledging the unique requirements of standalone or external parts of our project.

### **Improvement/Solution**
- Updated the lint configuration to exclude specific files from the 'color-no-hex' rule.
- Identified files where static hex color values are permissible due to their standalone nature or external usage.

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/23097

## **Manual testing steps**

1. Pull this branch
2. Run `yarn lint` 
3. Notice that there shouldn't be any static hex value warnings from stylelint

### **Before**
Below screencast shows warnings that should be ignored

https://github.com/MetaMask/metamask-extension/assets/8112138/6cd7d6ea-44da-4b7e-b594-931e9b22a007

### **After**
Below screencast shows no more stylelint warnings for static hex

https://github.com/MetaMask/metamask-extension/assets/8112138/a483fe16-eb82-46ae-8666-e62da418a95a

Linting still works as intended for other files

https://github.com/MetaMask/metamask-extension/assets/8112138/e2e43b34-ea6b-430b-8d2b-a7f662e30e0d

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues.
- [x] I've included manual testing steps.
- [x] I’ve included tests if applicable.
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable.
- [x] I’ve applied the right labels on the PR.
- [x] I’ve properly set the pull request status to "ready for review".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g., pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the issue it closes and includes the necessary testing evidence.

This PR template is filled out with a concise description of the changes, the motivation behind them, and the solution provided. It also includes a link to the related issue, outlines the manual testing steps, and adheres to the pre-merge checklists for both the author and the reviewer.

### CODE REVIEW

File: ui/css/utilities/colors.scss [hunk 1]
```diff
@@ -12,3 +17,4 @@
   // Required for the QR reader to work properly
   --qr-code-white-background: #fff;
 }
+/* stylelint-enable color-no-hex */
```
Comment (georgewrmarshall) [inline ui/css/utilities/colors.scss:20]: Remove this line
```suggestion
```

Comment (georgewrmarshall) [reply ui/css/utilities/colors.scss]: Updated https://github.com/MetaMask/metamask-extension/pull/23098/commits/bcec68c8bac52e2b5bf97d750b290e15b5f17d19

**Review by garrettbear — APPROVED** [review]

**Comment by HowardBraham** [comment]
This has been annoying me recently, so this is great!

But let's follow the conventions for giving a disable reason: https://eslint.org/docs/latest/use/configure/rules#comment-descriptions

**Comment by georgewrmarshall** [comment]
> This has been annoying me recently, so this is great!
> 
> But let's follow the conventions for giving a disable reason: https://eslint.org/docs/latest/use/configure/rules#comment-descriptions

Great suggestion but unfortunately as we've discovered individually that our version of stylelint doesn't support comments in this way. I'll create an issue to upgrade stylelint in the extension https://github.com/MetaMask/metamask-extension/issues/23099

Screenshot 2024-02-22 at 8 41 39 AM

**Review by brad-decker — APPROVED** [review]

**Review by HowardBraham — APPROVED** [review]

**Pull request merged**